.github/workflows/version-or-publish.yml

```diff
@@ -0,0 +1,85 @@
+name: Version or Publish Package
+
+on:
+  workflow_dispatch:
+    inputs:
+      version_tag:
+        required: true
+        description: The release version
+        type: string
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+permissions:
+  id-token: write
+  contents: write
+  attestations: write
+  actions: read
+
+jobs:
+  provenance:
+    uses: slsa-framework/slsa-github-generator/.github/workflows/builder_nodejs_slsa3.yml@v1.10.0
+    with:
+      run-scripts: "install-pnpm, install-deps, style, nx-build-skip-cache, nx-test-skip-cache, skip-lib-ignore"
+      node-version: "20.11.1"
+
+  release:
+    needs: [provenance]
+    name: Version or Publish
+    runs-on: ubuntu-latest
+    steps:
+      - name: Downlod Artifacts 
+        uses: actions/download-artifact@v3
```

Comment:
let's just hash here instead of `v3` you can use something like stepsecurity to identify the commit sha or manually you can get it from their repo